### PR TITLE
Added search bar to Asset Loader

### PR DESCRIPTION
**Pull request by aardschok**
* Implemented QLineEdit for searches.
* Text changed trigger `proxy.setFilterRegExp`
* Proxy set to `CaseInsensitive`

### CODE REVIEW

**Comment by mottosso** [comment]
This one actually contains #251, should we close one of them?

**Comment by BigRoy** [comment]
Close the other, because I believe this also has fixes for problems that the other one introduced. :)

File: avalon/tools/cbloader/app.py [hunk 1]
```diff
@@ -171,7 +171,8 @@ def _versionschanged(self):
         if active:
             rows = selection.selectedRows(column=active.column())
             if active in rows:
-                node = active.data(subsets.model.NodeRole)
+                node_data = subsets.model.NodeRole
+                node = active.data(node_data)
```
Comment (BigRoy) [inline avalon/tools/cbloader/app.py]: Personally I preferred the old style here: `node = active.data(subsets.model.NodeRole)`

File: avalon/tools/cbloader/delegates.py [hunk 2]
```diff
@@ -106,7 +106,9 @@ def displayText(self, value, locale):
 
 class VersionDelegate(QtWidgets.QStyledItemDelegate):
     """A delegate that display version integer formatted as version string.
-    
+
+    Note:
+        It is required to set source_model attribute.
```
Comment (BigRoy) [inline avalon/tools/cbloader/delegates.py]: This added docstring note isn't actually true so can be removed (that was only from some debugging in development)

File: avalon/tools/cbloader/widgets.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
 from .model import SubsetsModel
 from .delegates import PrettyTimeDelegate, VersionDelegate
 from . import lib
+from .proxy import FilterProxyModel
```
Comment (BigRoy) [inline avalon/tools/cbloader/widgets.py]: This import isn't used. This also means the added `proxy.py` file is redundant in `cbloader`.

**Comment by BigRoy** [comment]
@aardschok could you check the above comments made here and after changing test whether the functionality is all there? :) Then we should be safe to merge this.

**Comment by aardschok** [comment]
@BigRoy Everything works with said feedback implemented

**Comment by mottosso** [comment]
Would it be possible to make a small gif showcasing this new functionality? Ideally I'd like for all visual changes to come with a gif or still image.

**Comment by aardschok** [comment]
## Results:
asset_loader_sorted_by_version
asset_loader_filter_by_look
asset_loader_sorted_by_family

**Comment by mottosso** [comment]
Great! That'll make our future selves and any followers happy. :)

I try and do that to encourage comments without actually having to (1) clone, (2) checkout your PR branch, (3) set it up locally and (4) find a way to run it. A simple picture takes all that away.

**Comment by BigRoy** [comment]
Great work @aardschok!

**Pull request merged**